### PR TITLE
Deprecate main locale init

### PR DESCRIPTION
**Pull request by codebrainz**

### CODE REVIEW

File: plugins/demoplugin.c [hunk 1]
```diff
@@ -207,8 +207,6 @@ static void demo_cleanup(GeanyPlugin *plugin, gpointer data)
 
 void geany_load_module(GeanyPlugin *plugin, GModule *module, gint geany_api_version)
 {
-	/* main_locale_init() must be called for your package before any localization can be done */
-	main_locale_init(LOCALEDIR, GETTEXT_PACKAGE);
```
Comment (b4n) [inline plugins/demoplugin.c:211]: will this work?  there are `_()` calls before the `main_locale_init()` ones, so they probably won't work (well, I write "probably", but I believe "definitely")

**Comment by b4n** [comment]
Maybe to do such magic (which is probably nice -- IMO) there would rather need to be even higher level C macros for registering the plugin, maybe like

``` C

#if defined(GETTEXT_PACKAGE) && defined(LOCALEDIR)
# define GEANY_DEFINE_PLUGIN__i18n_preamble \
  { main_locale_init(LOCALEDIR, GETTEXT_PACKAGE); }
#else
# define GEANY_DEFINE_PLUGIN__i18n_preamble {}
#endif


#define GEANY_DEFINE_PLUGIN_WITH_CODE(p_name, p_desc, p_version, p_author, p_min_api_version, p_funcs, code) \
void geany_load_module(GeanyPlugin *plugin,                                   \
                       GModule     *module,                                   \
                       gint         api_version)                              \
{                                                                             \
  GeanyPluginFuncs funcs = p_funcs;                                           \
                                                                              \
  GEANY_DEFINE_PLUGIN__i18n_preamble                                          \
                                                                              \
  plugin->info->name = p_name;                                                \
  plugin->info->description = p_description;                                  \
  plugin->info->version = p_version;                                          \
  plugin->info->author = p_author;                                            \
                                                                              \
  if (GEANY_REGISTER_PLUGIN(plugin, p_min_api_version))                       \
  {                                                                           \
    code                                                                      \
  }                                                                           \
}


#define GEANY_DEFINE_PLUGIN(p_name, p_desc, p_version, p_author, p_min_api_version, p_funcs) \
  GEANY_DEFINE_PLUGIN_WITH_CODE(p_name, p_desc, p_version, p_author, p_min_api_version, p_funcs, {})


GEANY_DEFINE_PLUGIN(_("my plugin"), _("My wonderful plugin"),
                    "1.0", "John Doe <john@doe.com>",
                    225,
                    {
                      .init     = my_plugin_init,
                      .cleanup  = my_plugin_cleanup
                    })
```

**Comment by kugel-** [comment]
Are you serious?

**Comment by b4n** [comment]
halfway I guess :)

**Comment by codebrainz** [comment]
IIRC I had recommended some kind of macro like that during the mailing list discussions to make it simpler for the common case. Anyway, if my PR won't work, I'll just close it. We can entertain macro madness later after the `new_hooks` PR is merged :)

**Pull request closed without merging**